Source/CarthageKit/Xcode.swift

```diff
@@ -1015,6 +1009,18 @@ public func buildInDirectory(directoryURL: NSURL, withConfiguration configuratio
 			|> filter { projects in !projects.isEmpty }
 			|> flatMap(.Merge) { (projects: [(ProjectLocator, [String])]) -> SignalProducer<(String, ProjectLocator), CarthageError> in
 				return SignalProducer(values: projects)
+					|> flatMap(.Concat) { (project: ProjectLocator, schemes: [String]) in
+						// Only look for schemes that actually reside in the project
+						return SignalProducer(values: schemes)
+							|> filter { (scheme: String) -> Bool in
+								if let schemePath = project.fileURL.URLByAppendingPathComponent("xcshareddata/xcschemes/\(scheme).xcscheme").path {
+									return NSFileManager.defaultManager().fileExistsAtPath(schemePath)
+								}
+								return false
+							}
+							|> collect
+							|> map { (project, $0) }
+					}
```

Comment:
It seems like this `flatMap` is equivalent to a `map` using the Swift stdlib’s native `filter` method over `schemes`:

```swift
(project, filter(schemes) { scheme in
	project.fileURL.URLByAppendingPathComponent("xcshareddata/xcschemes/\(scheme).xcscheme").path != nil
})
```